core/contracts/financial-templates/implementation/PricelessPositionManager.sol

```diff
@@ -366,6 +376,31 @@ contract PricelessPositionManager is FeePayer {
         emit SettleExpiredPosition(msg.sender, totalRedeemableCollateral.rawValue, tokensToRedeem.rawValue);
     }
 
+    /**
+    * @notice Premature contract settlement under emergency circumstances.
+    * @dev Only the governor can call this function as they are permissioned within the `FinancialContractAdmin`.
+    * Upon emergency shutdown, the contract settlement time is set to the shutdown time. This enables withdrawal
+    * to occur via the standard settleExpired function call. Contract state is set to `ExpiredPriceRequested`
+    * which prevents re-entreaty into this function or the `expire` function.
+    */
+    function emergencyShutdown() external onlyPreExpiration() onlyOpenState() {
+        require(msg.sender == _getFinancialContractsAdminAddress());
+
+        contractState = ContractState.ExpiredPriceRequested;
+        // Expiratory time now becomes the current time (emergency shutdown time).
+        // Price requested at this time stamp. `settleExpired` can now withdraw at this timestamp.
+        uint oldExpirationTimestamp = expirationTimestamp;
+        expirationTimestamp = getCurrentTime();
+        _requestOraclePrice(expirationTimestamp);
```

Comment:
Why'd we decide not to add `_payFinalFees()` here?

Reply:
the main argument is that the entity that is calling this function (the govenor) is the same entity that would receive the final fee. It doesn't make sense for them to pay the final fee if they are getting it anyways.